src/redux/reducers/experimentSettings/processingConfig/loadedProcessingConfig.js

```diff
@@ -0,0 +1,49 @@
+/* eslint-disable no-param-reassign */
+import produce, { current } from 'immer';
+import _ from 'lodash';
+
+import initialState from '../initialState';
+
+// Will be removed once the default values are stored in the pipeline
+const withAutomaticSettings = (processingConfig) => {
+  const processingConfigWithAuto = _.cloneDeep(processingConfig);
+
+  // adding default config to every filter with auto option
+  Object.keys(processingConfig).forEach((step) => {
+    const currentStepSettings = processingConfig[step];
+    const sampleId = Object.keys(currentStepSettings).find(
+      (currentSample) => currentStepSettings[currentSample].auto,
+    );
+
+    if (sampleId) {
+      processingConfigWithAuto[step][sampleId]
+        .defaultFilterSettings = processingConfig[step][sampleId].filterSettings;
+    }
+  });
+
+  return processingConfigWithAuto;
+};
+
+const loadedProcessingConfig = produce((draft, action) => {
+  const { data, fromGem2s } = action.payload;
+
+  const originalProcessing = current(draft.processing);
+
+  let dataToSet = _.cloneDeep(data);
+
+  dataToSet.meta = {
+    ...originalProcessing.meta,
+    ...data.meta,
```

Comment:
This shouldn't be correct. It assumes that `originalProcessing.meta` is a simple object with no nested object properties that we'd need to merge. What if we end up with something like `stepsDone` here that needs special processing? Might be worth refactoring these two as a `_.merge` or something similar.